providers/update.rb

```diff
@@ -79,7 +79,7 @@ def assert_result(action, result)
 end
 
 def updates
-  @updates ||= [].tap do |updates|
+  @@updates ||= [].tap do |updates|
```

Comment:
using `run_context` should be better here than class variables, especially when chef run as a service daemon